packages/block-library/src/navigation/index.php

```diff
@@ -322,6 +372,29 @@ function block_core_navigation_get_fallback_blocks() {
 
 	$navigation_post = block_core_navigation_get_most_recently_published_navigation();
 
+	// If there are no navigation posts then try to find a classic menu.
```

Comment:
I would:
- make this whole new block a new function :) 
- put the contents of `block_core_navigation_get_classic_menu_fallback` into this new function (unlikely we'd reuse that)
- call this new function in `block_core_navigation_get_most_recently_published_navigation` as a case so that here we only call it once.

Reply:
I think we'll always need to call `block_core_navigation_get_most_recently_published_navigation` twice - once so we can determine whether there are any wp_navigation CPTs and then, if there aren't any but there is a classic navigation menu, to get the newly created one from the import.

Reply:
I did a bit of a refactor here nonetheless...